File: FieldOps.Domain/Entities/WorkOrder.cs
namespace FieldOps.Domain.Entities;

public class WorkOrder
{
    public int Id { get; set; }
    public int SiteId { get; set; }
    public string Title { get; set; } = string.Empty;
    public string? Description { get; set; }
    public string? SourceType { get; set; }
    public int? SourceId { get; set; }
    public int? AssignedTo { get; set; }
    public string Priority { get; set; } = "medium";
    public string Status { get; set; } = "open";
    public DateTime CreatedAt { get; set; }
    public DateTime? CompletedAt { get; set; }
}

File: client/src/types/workOrder.ts
export interface WorkOrder {
  id: number;
  siteId: number;
  siteName: string;
  title: string;
  description: string | null;
  sourceType: string | null;
  sourceId: number | null;
  assignedTo: number | null;
  assignedToName: string | null;
  priority: string;
  status: string;
  createdAt: string;
  completedAt: string | null;
}

export interface CreateWorkOrderRequest {
  siteId: number;
  title: string;
  description: string;
  priority: string;
  assignedTo: number | null;
}
